documentation/src/main/asciidoc/user_guide/server_protocols_rest.adoc

```diff
@@ -12,18 +41,129 @@ Note: TLS/ALPN with JDK8 requires additional steps from the client perspective.
 that you will need Jetty ALPN Agent or OpenSSL bindings.
 
 ==== REST API
-HTTP PUT and POST methods are used to place data in the cache, with URLs to address the cache name and key(s) - the data being the body of the request (the data can be anything you like). It is important that a Content-Type header is set. Other headers are used to control the cache settings and behaviour (detailed in that link).
+HTTP PUT and POST methods are used to place data in the cache, with URLs to address the cache name and key(s) - the data being the body of the request (the data can be anything you like). Other headers are used to control the cache settings and behaviour.
+
+===== Data formats
+
+====== Configuration
+
+Each cache exposed via REST stores data in a configurable data format defined by a https://en.wikipedia.org/wiki/Media_type[MediaType].
+
+An example of storage configuration is as follows:
+
+[source,xml]
+----
+<cache>
+   <encoding>
+      <key media-type="application/x-java-object; type=java.lang.Integer"/>
+      <value media-type="application/xml; charset=UTF-8"/>
+   </encoding>
+</cache>
+----
+
+When no MediaType is configured, Infinispan assumes "application/octet-stream" for both keys and values, with the following exceptions:
+
+* If the cache is indexed, it assumes "application/x-protostream"
+* If the cache is configured with compatibility mode, it assumes "application/x-java-object"
+
+====== Supported formats
+
+Data can be written and read in different formats than the storage format; Infinispan can convert between those formats when required.
+
+The following "standard" formats can be converted interchangeably:
+
+* _application/x-java-object_
+* _application/octet-stream_
+* _application/x-www-form-urlencoded_
+* _text/plain_
+
+The following formats can be converted to/from the formats above:
+
+* __application/xml__
+* _application/json_
+* _application/x-jboss-marshalling_
```

Comment:
Do we want to add formats for Kryo and ProtoStuff marshalling?

Reply:
I think they should be handled client side only for now, as those two are not part of the server